libarchive/entry.py

```diff
@@ -61,17 +61,36 @@ def isfifo(self):
 
     @property
     def islnk(self):
+        return bool(ffi.entry_hardlink_w(self._entry_p))
+
+    @property
+    def issym(self):
         return self.filetype & 0o170000 == 0o120000
 
     @property
+    def linkpath(self):
+        if self.islnk:
+            return ffi.entry_hardlink_w(self._entry_p)
+        if self.issym:
+            return ffi.entry_symlink_w(self._entry_p)
```

Comment:
> About efficiency I am not sure whether it is more efficient or not.

I'm sure, the current code calls `ffi.entry_hardlink_w` twice in the case of a hard link.